src/metrics.ts

```diff
@@ -87,11 +88,6 @@ export default class Metrics extends EventEmitter {
     this.timeout = timeout;
     this.resetBucket();
     this.httpOptions = httpOptions;
-
-    if (typeof this.metricsInterval === 'number' && this.metricsInterval > 0) {
```

Comment:
We should not start tasks in the constructor of a class, this reduces flexibility and may cause us to miss events because the using code is not able to register event listeners before the event actually happens.